library/item/item_extensions/src/main/java/org/quiltmc/qsl/item/extensions/api/bow/BowExtensions.java

```diff
@@ -43,6 +43,19 @@ public interface BowExtensions {
 	 * @param bowStack the item stack for the bow
 	 */
 	default @Range(from = 0, to = 1) float getCustomPullProgress(int useTicks, @NotNull ItemStack bowStack) {
```

Comment:
Hm, thinking about it, maybe it would be better if `BowItem.getPullProgress` was kept, and like, a `@ModifyConstant` to replace the `20.0F` constant with a `(float) this.getMaxDrawDuration()` was applied

Reply:
Not really possible, as that method only takes in the ticks used, so there is no way to specify the item. I think this is ok.